test/test_torch.py

```diff
@@ -18433,6 +18433,35 @@ def test_atleast(self, device, dtype):
         self._test_atleast_dim(torch.atleast_2d, np.atleast_2d, device, dtype)
         self._test_atleast_dim(torch.atleast_3d, np.atleast_3d, device, dtype)
 
+    def _test_special_stacks(self, torch_fn, np_fn, device, dtype):
```

Comment:
These tests are good, but this test's case generation is limited to replicating the same tensor shape for each element of the input list. Here are some cases I was thinking about:

- op(t) 
  - the behavior of `np.hstack(a)` is strange and `np.hstack(a) != np.hstack((a,))` (the same is true for `np.dstack`)
  - do we even support non-tuple arguments? if not we should validate this throws a runtime error
  - if we support single tensor arguments, is `np.hstack(a)`'s and `np.dstack(a)`'s behavior correct? 
- op((a, b, c, ...))
  - validating that if they differ on an unexpected dim an error is thrown (maybe `_test_special_stacks` should take a dim argument corresponding to the op?)
  - validating that if they differ only on the expected dim the result is equivalent to NumPy
-  are tensors with a size zero dim handled correctly? (if not that's OK, but let's assert it doesn't work)
-  `np.hstack` has special-handling of 1D tensors (as your implementation does), does `test_hstack` need a custom elaboration to test that behavior, in particular?
- validating that tensors with different shapes but the same post-`atleast_Xd` shapes meet the criteria work

For an example of the last bullet:

```
a = np.array([[[1],[2],[3]]])
b = np.array((4, 5, 6))
np.dstack((a, b))
: array([[[1, 4],
          [2, 5],
          [3, 6]]])
```

This is a good number of cases but validating each one by hand shouldn't be too laborious, I hope. 

What are your thoughts? Are there other cases I missed?